services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -74,7 +75,7 @@ public class RouterHttpRequestStats extends AbstractVeniceHttpStats {
 
   private final static Sensor totalInflightRequestSensor = localMetricRepo.sensor("total_inflight_request");
   static {
-    totalInflightRequestSensor.add("total_inflight_request_count", new Rate());
+    totalInflightRequestSensor.add(TOTAL_INFLIGHT_REQUEST_COUNT, new Gauge());
```

Comment:
`Gauge` doesn't honor any timed window, but just store the most recent value.
So the increment/decrement won't work.

If it is not windowed, who will update the metric to be `0` then?

Removing `record(-1)` should just work and if we want to speed up the stop, we should adjust time window size for this particular metric and currently, it is 10s, which means the Router can only be stopped after 20s since no traffic (by default, there are 2 windows).